zipkin/getmetrics_zipkin.py

```diff
@@ -0,0 +1,524 @@
+#!/usr/bin/env python
+import ConfigParser
+import collections
+import json
+import logging
+import os
+import re
+import socket
+import sys
+import time
+from optparse import OptionParser
+from itertools import islice
+from datetime import datetime
+from statistics import median
+import urlparse
+import httplib
+import requests
+
+'''
+This script gathers data to send to Insightfinder
+'''
+
+
+def start_data_processing():
+    """ get traces from the last <samplingInterval> minutes """
+    # fork off a process to handle metric agent
+    parent = os.fork()
+    if parent > 0:
+        track['mode'] = "LOG"
+        if_config_vars['projectName'] += '-log'
+        logger.debug(str(os.getpid()) + ' is running the log agent')
+    else:
+        track['mode'] = "METRIC"
+        if_config_vars['projectName'] += '-metric'
+        logger.debug(str(os.getpid()) + ' is running the metric agent')
+    timestamp_fixed = int(time.time() * 1000)
+    traces = zipkin_get_traces()
+    for trace in traces:
+        for span in trace:
+            process_zipkin_span(timestamp_fixed, span)
+
+
+def process_zipkin_span(timestamp_fixed, span):
+    """
+    for a single span, check if it passes the filter, add it to the
+    current chunk, and send the chunk if appropriate
+    """
+    endpoint = span['localEndpoint']
+    span_name = span['name']
+    if filter_zipkin_span(endpoint['serviceName'], span_name):
+        return
+
+    timestamp = int(span.pop('timestamp')) / 1000
+    instance = make_instance_name(endpoint)
+
+    if track['mode'] == "LOG":
+        log_handoff(timestamp, instance, span)
+    else:
+        metric_handoff(timestamp_fixed, span_name, instance, span['duration'])
+
+
+def filter_zipkin_span(service_name, span_name):
+    """ determine if this span should be excluded """
+    if len(agent_config_vars['services_filter']) != 0 and service_name not in agent_config_vars['services_filter']:
```

Comment:
'in' means the partially match. please confirm this is what you want

Reply:
yes, should they want to do partial matching.